packages/fast-components-react-msft/src/text-action/text-action.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    ButtonPosition,
+    TextActionHandledProps,
+    TextActionProps,
+    TextActionUnhandledProps,
+} from "./text-action.props";
+import { textFieldOverrides } from "@microsoft/fast-components-styles-msft";
+import { TextField } from "../text-field";
+import { get } from "lodash-es";
+
+class TextAction extends Foundation<
+    TextActionHandledProps,
+    TextActionUnhandledProps,
+    {}
+> {
+    public static displayName: string = "TextAction";
+
+    public static defaultProps: Partial<TextActionProps> = {
+        buttonPosition: ButtonPosition.after,
+    };
+
+    protected handledProps: HandledProps<TextActionHandledProps> = {
+        afterGlyph: void 0,
+        beforeGlyph: void 0,
+        button: void 0,
+        buttonPosition: void 0,
+        managedClasses: void 0,
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        return (
+            <div className={this.generateClassNames()}>
+                {this.getButton() && this.props.buttonPosition === ButtonPosition.before
+                    ? this.props.button(
+                          get(this.props, "managedClasses.textAction_button"),
+                          this.props.disabled
+                      )
+                    : null}
+                {this.generateBeforeGlyph()}
+                <TextField
```

Comment:
If we type into a TextAction with a button and hit `enter`, does it use the button action or the TextField's?

Reply:
As it is, it uses TextField's

Reply:
Not sure if this is what I'd expect as a developer, but I guess its easy enough to code in with `preventDefault`

Reply:
Typically this would be wrapped in a `form` that would handle the action - or other props could be passed to the button

Reply:
Open to a discussion on best implementation. I know we are trying to get away from prescribing to heavily in situations like this and allowing the developers to wire things up how they need in their instance.

Reply:
Resolving for now since there has been no follow-up.